src/bluesky/callbacks/tiled_writer.py

```diff
@@ -87,12 +88,13 @@ def _convert_resource_to_stream_resource(self, doc: Union[Resource, StreamResour
                 raise RuntimeError(f"`Resource` or `StreamResource` legacy document is missing a '{expected_key}'")
 
         # Convert the Resource (or old StreamResource) document to a StreamResource document
-        doc["mimetype"] = MIMETYPE_LOOKUP[doc.pop("spec")]
-        doc["parameters"] = doc.pop("resource_kwargs", {})
-        file_path = doc.pop("root").strip("/") + "/" + doc.pop("resource_path").strip("/")
-        doc["uri"] = "file://localhost/" + file_path
+        resource_dict = cast(dict, doc)
+        stream_resource_doc["mimetype"] = MIMETYPE_LOOKUP[resource_dict.pop("spec")]
+        stream_resource_doc["parameters"] = resource_dict.pop("resource_kwargs", {})
+        file_path = resource_dict.pop("root").strip("/") + "/" + resource_dict.pop("resource_path").strip("/")
+        stream_resource_doc["uri"] = "file://localhost/" + file_path
```

Comment:
@evalott100 This was the only way I could get this part to work nicely, let me know if you can think of something better!

Reply:
Nah that's a good way to do it!